tooling-docker/src/main/java/org/jboss/eap/qe/ts/common/docker/Docker.java

```diff
@@ -0,0 +1,204 @@
+
+    protected void start() throws Exception {
+        List<String> cmd = new ArrayList<>();
+
+        cmd.add("docker");
+        cmd.add("run");
+        cmd.add("--name");
+        cmd.add(uuid);
+
+        for (String port : ports) {
+            cmd.add("-p");
+            cmd.add(port);
+        }
+
+        for (Map.Entry<String, String> envVar : environmentVariables.entrySet()) {
+            cmd.add("-e");
+            cmd.add(envVar.getKey() + "=" + envVar.getValue());
+        }
+
+        cmd.add(image);
+
+        cmd.addAll(commandArguments);
+
+        System.out.println(Ansi.ansi().reset().a("Starting container ").fgCyan().a(name).reset()
+                .a(" with ID ").fgYellow().a(uuid).reset());
+
+        Process dockerRunProcess = new ProcessBuilder()
+                .redirectErrorStream(true)
+                .command(cmd)
+                .start();
+
+        outputPrinter.execute(() -> {
+            try (BufferedReader reader = new BufferedReader(new InputStreamReader(dockerRunProcess.getInputStream()))) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    System.out.println(Ansi.ansi().fgCyan().a(name).reset().a("> ").a(line));
+                }
+            } catch (IOException ignored) {
+                // ignore as any stop of docker container breaks the reader stream
+                // note that shutdown of docker would be already logged
+            }
+        });
+
+        containerReadyCondition.waitUntilReady(containerReadyTimeout);
+    }
+
+    protected void stop() throws Exception {
+        System.out.println(Ansi.ansi().reset().a("Stopping container ").fgCyan().a(name).reset()
+                .a(" with ID ").fgYellow().a(uuid).reset());
+
+        new ProcessBuilder()
+                .command("docker", "stop", uuid)
+                .start()
+                .waitFor(10, TimeUnit.SECONDS);
+
+        outputPrinter.shutdown();
+        outputPrinter.awaitTermination(10, TimeUnit.SECONDS);
+
+        new ProcessBuilder()
+                .command("docker", "rm", uuid)
+                .start()
+                .waitFor(10, TimeUnit.SECONDS);
+    }
+
+    @Override
+    protected void before() throws Throwable {
+        start();
+    }
+
+    @Override
+    protected void after() {
+        try {
+            stop();
+        } catch (Exception e) {
+            System.out.println(Ansi.ansi().reset().a("Failed stopping container ").fgCyan().a(name).reset()
+                    .a(" with ID ").fgYellow().a(uuid).reset());
+        }
+    }
+
+    public static class Builder {
+        private String uuid;
+        private String name;
+        private String image;
+        private List<String> ports = new ArrayList<>();
+        private Map<String, String> environmentVariables = new HashMap<>();
+        private List<String> commandArguments = new ArrayList<>();
+        private long containerReadyTimeoutInMillis = 60_000; // 1 minutes
```

Comment:
This means that docker has hard limit 1 minute to start (including images download) ?

Reply:
Yes, it's including download+docker container start. It's default value which can be changed by setContainerReadyTimeout(timeout) . It's expected that it will be set properly for each container. 

Do you want to set different default timeout? Like 5 min for example to be more safe.

Reply:
no timeout change, just wanted to know the details